ui/mirage/handlers/sync.js

```diff
@@ -90,6 +90,14 @@ export default function (server) {
         connection_details,
         name,
         type,
+        /* 
+        these only exist if a delete has been initiated, including here for dev/testing purposes
+        if only `purge_initiated_at` present the delete progress banner renders
+        if `purge_error` also has a value then delete failed banner renders
+        */
+        purge_initiated_at: '2024-01-09T16:54:28.463879-07:00',
+        // WIP (backend hasn't added yet) update when we have a realistic error message)
+        // purge_error: '1 error occurred: association could for some confusing reason not be un-synced!',
```

Comment:
To make this more dynamic we could add both `purge_initiated_at` and `purge_error` to the factory and then when a delete request comes in, update the record in mirage by setting those properties instead of removing. No need to go down that path now but just mentioning for the future if we need it for testing.

Reply:
Yeah I started doing that last night and ran out of brain power. Definitely a good improvement.